Extensions/System/StringResources/Source/netfx-System.StringResources.targets

```diff
@@ -23,112 +23,133 @@
 	SOFTWARE.
 -->
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <UsingTask TaskName="NetFx.StringResources" AssemblyFile="netfx-System.StringResources.dll" />
-
-  <PropertyGroup>
-    <CoreCompileDependsOn>
-      CollectResxCode;
-      GenerateStringResources;
-      CompileStringResources;
-      $(CoreCompileDependsOn)
-    </CoreCompileDependsOn>
-  </PropertyGroup>
-
-  <ItemDefinitionGroup>
-    <ResxCode>
-      <!-- Whether the generated strong-typed class should be public or not.
+	<UsingTask TaskName="NetFx.StringResources" AssemblyFile="netfx-System.StringResources.dll" />
+
+	<PropertyGroup>
+		<GenerateStringResources Condition="'$(GenerateStringResources)' == ''">true</GenerateStringResources>
+		<StringResourceItemsPath Condition="'$(StringResourceItemsPath)' == ''">$(IntermediateOutputPath)Strings.Resources.items</StringResourceItemsPath>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<PrepareResourcesDependsOn>
+			GenerateStringResources;
+			$(PrepareResourcesDependsOn)
+		</PrepareResourcesDependsOn>
+
+		<CoreCompileDependsOn>
+			GenerateStringResources;
+			$(CoreCompileDependsOn)
+		</CoreCompileDependsOn>
+
+		<ResGenDependsOn>
+			_AugmentLinkedResources;
+			$(ResGenDependsOn)
+		</ResGenDependsOn>
+	</PropertyGroup>
+
+	<ItemDefinitionGroup>
+		<ResxCode>
+			<!-- Whether the generated strong-typed class should be public or not.
            Defaults to whatever the corresponding .resx file specifies. -->
-      <Public></Public>
-      <!-- Customize the generated class name to be something other than 'Strings' -->
-      <TargetClassName></TargetClassName>
-      <Generator>MSBuild:Compile</Generator>
-    </ResxCode>
-  </ItemDefinitionGroup>
-
-  <PropertyGroup>
-    <ResGenDependsOn>GenerateMissingResources;$(ResGenDependsOn)</ResGenDependsOn>
-  </PropertyGroup>
-
-  <Target Name="CollectResxCode"
-          BeforeTargets="CoreCompile"        
-          Inputs="@(EmbeddedResource)" 
-          Outputs="@(EmbeddedResource->'%(Identity)-batch')" 
-          DependsOnTargets="PrepareResources" 
+			<Public></Public>
+			<!-- Customize the generated class name to be something other than 'Strings' -->
+			<TargetClassName></TargetClassName>
+			<Generator>MSBuild:Compile</Generator>
+		</ResxCode>
+	</ItemDefinitionGroup>
+
+	<Target Name="GenerateStringResources" Condition="'$(GenerateStringResources)' == 'true' and '$(_GeneratingStringResources)' == ''">
+
+		<MSBuild Projects="$(MSBuildProjectFullPath)"
+						 Targets="_GenerateStringResources"
+						 Properties="_GeneratingStringResources=true;BuildProjectReferences=false" />
```

Comment:
Why do we need to shell out to another MSBuild process? This is quite heavyweight: 
<img width="911" alt="00001753-1" src="https://user-images.githubusercontent.com/264396/31621565-ee46a214-b291-11e7-9fbc-a523d9daf5ec.png">